mm2src/coins/lightning.rs

```diff
@@ -206,14 +165,70 @@ impl LightningCoin {
             .pay_pubkey(destination, payment_preimage, amount_msat, final_cltv_expiry_delta)
             .map_to_mm(|e| SendPaymentError::PaymentError(format!("{:?}", e)))?;
         let payment_hash = PaymentHash(Sha256::hash(&payment_preimage.0).into_inner());
+        let payment_type = PaymentType::OutboundPayment {
+            destination: Some(destination),
```

Comment:
In what case can the destination be `None`?

Reply:
It can be `None` if the public key is included in the invoice signature, that's why the `payee_pub_key` method returns an option https://github.com/KomodoPlatform/atomicDEX-API/blob/f1001a0bea615cd533b39f768842845a6b55c7ca/mm2src/coins/lightning.rs#L130. But in this case, I guess I can use the provided [`recover_payee_pub_key`](https://github.com/lightningdevkit/rust-lightning/blob/2a3d0f7ef9843e3c547c3e6a52f1bf261a22c413/lightning-invoice/src/lib.rs#L710) method to recover the pubkey. Will fix the `destination` to not be an option.